torch/distributed/elastic/supervisor/hostmanager.py

```diff
@@ -0,0 +1,354 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import ctypes
+import io
+import logging
+import os
+import pickle
+import signal
+import socket
+import subprocess
+import sys
+import time
+import traceback
+from contextlib import nullcontext
+from pathlib import Path
+from random import random
+from typing import Callable, Dict, List, Mapping, Optional
+
+import zmq  # type: ignore[import-not-found]
+
+from torch.distributed.elastic.supervisor import (
+    HEARTBEAT_INTERVAL,
+    HEARTBEAT_LIVENESS,
+    ProcessFailedToStart,
+)
+from torch.distributed.elastic.utils.logging import get_logger
+
+logger: logging.Logger = get_logger(__name__)
+
+ABORT_INTERVAL = 5
+__NR_pidfd_open = 434
```

Comment:
This hardcodes some behavior for x86 + linux (syscall numbers differ between x86 and ARM). Don't you want to add assert of sort that this code is indeed only imported from those combination of OS+HW ?